### PR TITLE
Add AstraDB document loader

### PR DESCRIPTION
**Pull request by cbornet**
- **Description:** this adds the AstraDB document loader and an integration test
  - **Twitter handle:** cbornet_

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
Great !

**Comment by eolivelli** [comment]
@hemidactylus can you please take a look ?

**Comment by cbornet** [comment]
Added support for sort option.

**Review by hwchase17 — APPROVED** [review]
thanks!

File: libs/community/tests/integration_tests/document_loaders/test_astradb.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+Test of Astra DB document loader class `AstraDBLoader`
+
+Required to run this test:
+    - a recent `astrapy` Python package available
+    - an Astra DB instance;
+    - the two environment variables set:
+        export ASTRA_DB_API_ENDPOINT="https://<DB-ID>-us-east1.apps.astra.datastax.com"
+        export ASTRA_DB_APPLICATION_TOKEN="AstraCS:........."
+    - optionally this as well (otherwise defaults are used):
+        export ASTRA_DB_KEYSPACE="my_keyspace"
```
Comment (msmygit) [inline libs/community/tests/integration_tests/document_loaders/test_astradb.py:11]: should we use `namespace` notation (as opposed to _keyspace_) at here, on lines 23, 37, 60 & 89 too just to be consisten everywhere?

Comment (cbornet) [reply libs/community/tests/integration_tests/document_loaders/test_astradb.py:11]: The name `ASTRA_DB_KEYSPACE ` has been used everywhere for the env var in astrapy, LangChain and RAGStack. I'm OK to change it but then we should change it everywhere. @hemidactylus ?

File: libs/community/langchain_community/document_loaders/astradb.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                "Could not import a recent astrapy python package. "
+                "Please install it with `pip install --upgrade astrapy`."
+            )
+
+        # Conflicting-arg checks:
+        if astra_db_client is not None:
+            if token is not None or api_endpoint is not None:
+                raise ValueError(
+                    "You cannot pass 'astra_db_client' to AstraDB if passing "
+                    "'token' and 'api_endpoint'."
```
[the remark this answers is not in the document]
Comment (cbornet) [reply libs/community/langchain_community/document_loaders/astradb.py:43]: Indeed.

**Pull request merged**